lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -245,11 +245,11 @@ def request_params_hosts(params)
       end
     end
 
-    class ExportCommand < HammerCLIKatello::SingleResourceCommand
+    class LegacyExportCommand < HammerCLIKatello::SingleResourceCommand
       include HammerCLIForemanTasks::Async
 
       action :export
-      command_name "export"
+      command_name "export-legacy"
```

Comment:
Normally true, this is a tough case where we want export to be the new workflow, but give users access to the old workflow. I could not think of a better way to do this other than renaming the old command to prevent carrying around some weird command for new functionality. Also, I still haven't figure dout how to "deprecate" a command.

Reply:
This is the same action as the old export command. Is it not possible to make them the same command, differentiating behavior with some flag/option? This would be the first hammer command to duplicate an existing API endpoint, a red flag in my opinion.

Reply:
Since the old export is allowed from a remote hammer machine, perhaps the flag could be something like `--remote` for the old command or `--local` for the new command (or something similar with more detail; naming is hard).

Reply:
I think you are thinking in the right idea, but conceptually it puts us in a place we don't want to be. We aren't looking to support remote and local and have them two completely different processes. We want to move users to this new way of doing export and import as the only way we support doing it. With that in mind, any other thoughts how we could differentiate?

Reply:
> We aren't looking to support remote and local and have them two completely different processes

Great, it sounds like we want to support remote and local and have them be the same process (i.e. with `hammer content-view version export`). This PR currently fails to export when running the hammer command from a different box. Were you planning to support (or handle) this? There need not be a flag for something we can determine programmatically.

Reply:
Let me clarify. We want to support only the new method for export and import. That method requires local hammer only. There is no support for running hammer remotely to perform this new export and import.

Reply:
> That method requires local hammer only

The first hammer command in hammer-cli-katello of its kind afaik. Perhaps this remote CLI tool isn't the best place to put local-only export (i.e. hammer-cli-foreman-admin is local only).

Again, you must at least handle the remote case because hammer-cli-katello is a remote CLI tool. All I'm suggesting is that you handle it by invoking the old export functionality, which is the first step of your new functionality anyway (just the API call).

Reply:
The old export is entirely different than the new export in terms of how its designed to work. We are deprecating that API to enforce this as well. This is Katello functionality however, so spinning up an entire hammer project just to handle this seems like overkill. There is no clean way to provide "remote" support for this functionality. So I think we either have this special snowflake or we don't provide users a simple, standard orchestrated way to export/import content views. The alternative is to push all of this functionality to Ansible, however, we have to solve a lot of deliverable questions if we do that.

Reply:
> We are deprecating that API to enforce this as well

Don't you rely on that API for the new export command? Where's the Katello PR to deprecate the API?

Reply:
We do not use that export API at all for new export/import. On the to do list to deprecate it.